datafusion/core/src/physical_optimizer/sort_enforcement2.rs

```diff
@@ -0,0 +1,2872 @@
+        // parent requirements are more specific, adjust the request child requirements and push down the new requirements
+        let adjusted = parent_required.map(|r| r.to_vec());
+        RequirementsCompatibility::Compatible(adjusted)
+    } else {
+        RequirementsCompatibility::NonCompatible
+    }
+}
+
+/// Compares window expression's `window_request` and `parent_required_expr` ordering, returns
+/// whether we should reverse the window expression's ordering in order to meet parent's requirements.
+fn check_alignment(
+    input_schema: &SchemaRef,
+    window_request: &PhysicalSortRequirements,
+    parent_required_expr: &PhysicalSortRequirements,
+) -> bool {
+    if parent_required_expr.expr.eq(&window_request.expr)
+        && window_request.sort_options.is_some()
+        && parent_required_expr.sort_options.is_some()
+    {
+        let nullable = parent_required_expr.expr.nullable(input_schema).unwrap();
+        let window_request_opts = window_request.sort_options.unwrap();
+        let parent_required_opts = parent_required_expr.sort_options.unwrap();
+        if nullable {
+            window_request_opts == reverse_sort_options(parent_required_opts)
+        } else {
+            // If the column is not nullable, NULLS FIRST/LAST is not important.
+            window_request_opts.descending != parent_required_opts.descending
+        }
+    } else {
+        false
+    }
+}
+
+fn reverse_window_sort_requirements(
+    request_child: Option<&[PhysicalSortRequirements]>,
+) -> Option<Vec<PhysicalSortRequirements>> {
+    request_child.map(|request| {
+        request
+            .iter()
+            .map(|req| match req.sort_options {
+                None => req.clone(),
+                Some(ops) => PhysicalSortRequirements {
+                    expr: req.expr.clone(),
+                    sort_options: Some(reverse_sort_options(ops)),
+                },
+            })
+            .collect::<Vec<_>>()
+    })
+}
+
+/// Whether to reverse the top WindowExec's sort requirements.
+/// Considering the requirements of the descendants WindowExecs and leaf nodes' output ordering.
+/// TODO！considering all the cases
+fn should_reverse_window_sort_requirements(
+    window_plan: Arc<dyn ExecutionPlan>,
+    top_requirement: Option<&[PhysicalSortRequirements]>,
+    top_reversed_requirement: Option<&[PhysicalSortRequirements]>,
+) -> bool {
+    if top_requirement.is_none() {
+        return false;
+    }
+    let WindowExecInfo { window_expr, .. } =
+        extract_window_info_from_plan(&window_plan).unwrap();
+    let reverse_window_expr = window_expr
+        .iter()
+        .map(|e| e.get_reverse_expr())
+        .collect::<Option<Vec<_>>>();
+    if reverse_window_expr.is_none() {
+        return false;
+    }
+    let flags = window_plan
+        .children()
+        .into_iter()
+        .map(|child| {
+            // If the child is leaf node, check the output ordering
+            if child.children().is_empty()
```

Comment:
At this state we know that executor is `window`.  It has exactly one child. Hence I think we can remove this iteration. We can just use body of `map`, where `child` is `window_plan.children()[0].clone()`.